src/main/java/net/minecraftforge/common/util/ITeleporter.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2019.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.common.util;
+
+import java.util.function.Function;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.world.Teleporter;
+import net.minecraft.world.dimension.Dimension;
+import net.minecraft.world.server.ServerWorld;
+
+/**
+ * Interface for handling the placement of entities during dimension change.
+ *
+ * An implementation of this interface can be used to place the entity
+ * in a safe location, or generate a return portal, for instance.
+ *
+ * See the {@link net.minecraft.world.Teleporter} class, which has
+ * been patched to implement this interface, for a vanilla example.
+ */
+public interface ITeleporter {
+
+    /**
+     * Called to handle placing the entity in the new world.
+     *
+     * The initial position of the entity will be its
+     * position in the origin world, multiplied horizontally
+     * by the computed cross-dimensional movement factor
+     * (see {@link Dimension#getMovementFactor()}).
+     *
+     * Note that the supplied entity has not yet been spawned
+     * in the destination world at the time.
+     *
+     * @param entity           the entity to be placed
+     * @param currentWorld     the entity's origin
+     * @param destWorld        the entity's destination
+     * @param yaw              the suggested yaw value to apply
+     * @param repositionEntity a function to reposition the entity, which returns the new entity in the new dimension. This is the vanilla implementation of the dimension travel logic. If the supplied boolean is true, it is attempted to spawn a new portal.
+     */
+    default Entity placeEntity(Entity entity, ServerWorld currentWorld, ServerWorld destWorld, float yaw, Function<Boolean, Entity> repositionEntity) {
```

Comment:
You don't describe what the return should be, so the docs are incomplete.

Reply:
I fixed this.